### DISCUSSION TITLE
PV values too small and/or "inverted"?

### DISCUSSION THREAD

**Discussion by badnetmask** [carlosedp/cluster-monitoring#147] [General]
Hello all,

First of all, let me thank the developer and the maintainers for this great project.

Today I faced an issue where the Prometheus PV filled out and stopped recording metrics. Looking ar `vars.jsonnet`, I have a feeling like the PV sizes are too small and/or "inverted":

https://github.com/carlosedp/cluster-monitoring/blob/9b70233e4ef10768db99e18f8539fcae09fc0e2e/vars.jsonnet#L76-L77

With merely 2 days of operation on a 2 node cluster, the 2Gi Prometheus PV filled completely, while the 20Gi Grafana is only using 450Mi.

In other people's experiences, what would be good values for the PVs here?

**Comment by carlosedp** [comment]
The default values in the config are samples or placeholders for small test clusters. This highly depends on the amount of nodes and data generated by your applications. I'd go with something like 10-20Gb for Prometheus. The Grafana PV can be reduced to 2-4Gb and might never get filled.

**Reply by badnetmask** [reply]
Thanks, I did "invert" by myself, but I did not have enough time go do dig deep into the configs. Is there some sort of rotation/retention time in the Prometheus configs?

**Comment by carlosedp** [comment]
by default Prometheus retains data for 15 days. You can config this in vars.jsonnet in the `retention` parameter.
